src/components/SettingsModal.js

```diff
@@ -0,0 +1,107 @@
+import React, { useState, useContext, useEffect } from "react";
+import reactDom from "react-dom";
+import classnames from "classnames";
+import { v4 as uuidv4 } from "uuid";
+import { ModalContext } from "./ClockFace.js"
+import "./SettingsModal.scss";
+
+const SettingsModal = () => {
+  const { isModalOpen, setIsModalOpen, sessions, setSessions } = useContext(ModalContext);
+  const [localSession, setLocalSession] = useState(sessions);
+  const [hasSettingsChanged, setHasSettingsChanged] = useState(false);
+
+  useEffect(() => {
+    if (isModalOpen) {
+      setLocalSession(sessions);
+      setHasSettingsChanged(false);
+    }
+  }, [isModalOpen, sessions])
+
+  if (!isModalOpen) {
+    return null;
+  }
+
+  const applyAndClose = () => {
+    setSessions(localSession)
+    setIsModalOpen(false);
+  }
+
+  const handleInputChange = (e) => {
+    if (e.target.valueAsNumber < 1 || e.target.valueAsNumber > 60 || !e.target.valueAsNumber) {
+      return false;
+    }
+    else {
+      setHasSettingsChanged(true);
+      if (e.target.name === "longSession") {
+        const newSettings = [...localSession];
+        newSettings[0] = e.target.valueAsNumber * 60;
+        newSettings[2] = e.target.valueAsNumber * 60;
+        setLocalSession(newSettings);
+      }
+      else if (e.target.name === "shortSession") {
+        const newSettings = [...localSession];
+        newSettings[1] = e.target.valueAsNumber * 60;
+        newSettings[3] = e.target.valueAsNumber * 60;
+        setLocalSession(newSettings);
+      }
+    }
+  }
+
+  const formatToMinutes = (seconds) => {
+    return Math.floor(seconds / 60);
+  }
+
+  return reactDom.createPortal(
+    <div className="SettingsModal__overlay">
+      <div className="SettingsModal">
+        <h2>Settings</h2>
+        <div className="SettingsModal__form">
+          <label>
+            Work:
+          </label>
+          <input
+            key={uuidv4()}
+            type="number"
+            max="60"
+            min="1"
+            name="longSession"
+            value={formatToMinutes(localSession[0])}
+            onChange={e => handleInputChange(e)}
+          >
+          </input>
+          <label>
+            Break:
+          </label>
+          <input
+            key={uuidv4()}
+            type="number"
+            max="60"
+            min="1"
+            name="shortSession"
+            value={formatToMinutes(localSession[1])}
+            onChange={e => handleInputChange(e)}
+          >
+          </input>
+        </div>
+        <div className="SettingsModal__modal-controls">
+          <button
+            className={classnames("apply-close-button", { hasSettingsChanged })}
```

Comment:
I think this button's className is "deep" enough? `SettingsModal__modal-controls__apply-close-button`seems to be too long? And `SettingsModal__apply-close-button` defies the purpose of SASS.

Reply:
It's really up to you on how you want to do things. To me `SettingsModal__apply-close-button` is good because there's only one selector and you can simply add style in scss file like this:
```scss
.SettingsModal {
  &__apply-close-button
}
```
which still makes it work.

You can also just use what you have now and remember to scope it under `SettingsModal` class. This is totally fine. 

One thing to keep in mind is you don't want to nested too many layer in your scss file because scss will basically create a huge class list when compiled.
```sass
.A {
  .B {
    .C {
      .D {
        background: red;
      }
      .E {
        background: green;
      }
    }
  }
}

// will get compiled to this
.A .B .C .D {
  background: red;
}

.A .B .C .E {
  background: green;
}

// three to four layer is still okay but imaging if you have 5 or 6 layer and basically the more layer the less performance for the css
```